### PR TITLE
feat: check available snx withdraw liquidity

### PR DESCRIPTION
**Pull request by artsiomYavorski**

### CODE REVIEW

File: packages/trading-widget/src/core-kit/hooks/trading/synthetix-v3/use-available-withdraw-liquidity.ts [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+      contractGuardData?.[0]?.result ?? BigInt(0),
+      contractGuardData?.[1]?.result?.[1] ?? AddressZero,
+    ],
+    query: {
+      enabled:
+        !!contractGuardAddress &&
+        contractGuardData?.every(({ result }) => !!result),
+      staleTime: Infinity,
+      select: (data) => {
+        const value = normalizeNumber(data.toString())
```
Comment (dimlbc) [inline packages/trading-widget/src/core-kit/hooks/trading/synthetix-v3/use-available-withdraw-liquidity.ts:83]: @D-Ig 
Will `getAccountAvailableCollateral` return amount always in 18 decimals?

File: packages/trading-widget/src/core-kit/hooks/trading/synthetix-v3/use-withdraw-liquidity.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+      ?.fractionUsdNumber ?? 0
+
+  return {
+    noLiquidity: new BigNumber(availableLiquidity?.value ?? '0')
+      .multipliedBy(
+        normalizeNumber(
+          availableLiquidityAssetPriceUsd?.toString() ?? '0',
+          SYNTHETIX_V3_VAULTS_WITHDRAW_ASSET_DECIMALS_MAP[
+            availableLiquidity?.address ?? AddressZero
+          ],
```
Comment (dimlbc) [inline packages/trading-widget/src/core-kit/hooks/trading/synthetix-v3/use-withdraw-liquidity.ts:54]:
```suggestion
```
Price is always in 18 decimals, I believe we can remove `SYNTHETIX_V3_VAULTS_WITHDRAW_ASSET_DECIMALS_MAP` in that case

File: packages/trading-widget/src/core-kit/hooks/trading/synthetix-v3/use-withdraw-liquidity.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        ),
+      )
+      .lt(snxFractionUsd),
+    symbol:
+      SYNTHETIX_V3_VAULTS_WITHDRAW_ASSET_SYMBOL_MAP[
+        availableLiquidity?.address ?? AddressZero
+      ],
```
Comment (dimlbc) [inline packages/trading-widget/src/core-kit/hooks/trading/synthetix-v3/use-withdraw-liquidity.ts:61]:
```suggestion
    symbol:
      SYNTHETIX_V3_VAULTS_WITHDRAW_ASSET_SYMBOL_MAP[
        address
      ],
```
We need to use vault address here as map includes `[vaultAddress]: symbol`

**Comment by dimlbc** [comment]
@artsiomYavorski Also fix unit tests 🙏

File: packages/trading-widget/src/core-kit/hooks/trading/synthetix-v3/use-available-withdraw-liquidity.ts (hunk 1 above)
Comment (D-Ig) [reply packages/trading-widget/src/core-kit/hooks/trading/synthetix-v3/use-available-withdraw-liquidity.ts:83]: > @D-Ig Will `getAccountAvailableCollateral` return amount always in 18 decimals?

yes https://github.com/dhedge/dhedge-v2/blob/master/contracts/interfaces/synthetixV3/ICollateralModule.sol#L53

File: packages/trading-widget/src/core-kit/hooks/trading/synthetix-v3/use-withdraw-liquidity.ts (hunk 2 above)
Comment (artsiomYavorski) [reply packages/trading-widget/src/core-kit/hooks/trading/synthetix-v3/use-withdraw-liquidity.ts]: removed

File: packages/trading-widget/src/core-kit/hooks/trading/synthetix-v3/use-withdraw-liquidity.ts (hunk 3 above)
Comment (artsiomYavorski) [reply packages/trading-widget/src/core-kit/hooks/trading/synthetix-v3/use-withdraw-liquidity.ts]: changed to `address`

**Review by dimlbc — APPROVED** [review]

**Pull request merged**